src/olympia/bandwagon/views.py

```diff
@@ -676,9 +677,22 @@ def get_account_viewset(self):
         return self.account_viewset
 
     def get_queryset(self):
-        return Collection.objects.filter(
+        qs = Collection.objects.filter(
             author=self.get_account_viewset().get_object()).order_by(
             '-modified')
+        if 'has_addon' in self.request.GET and self.action == 'list':
+            # When listing collections belonging to someone, you can pass an
+            # addon id and we'll expose whether or not the add-on is part of
+            # each collection returned. No checks are made to see whether the
+            # add-on is public, but we're only revealing that it's part of a
+            # collection, and you need to be the collection author or an admin
+            # to access this anyway.
```

Comment:
ugh, logic all over the place.  I don't have a better way to implement it though.